python/tank/platform/engine.py

```diff
@@ -192,6 +194,10 @@ def __init__(self, tk, context, engine_instance_name, env):
         qt_abstraction.QtCore = qt.QtCore
         qt_abstraction.QtGui = qt.QtGui
 
+        # load the fonts. this will work if there is a QApplication instance
+        # available.
+        self._ensure_core_fonts_loaded()
```

Comment:
My thought here was that we have several different scenarios already in terms of how the `QApplication` is actually started (we can't load the fonts until there is a `QApplication`). 

* standard DCCs (Maya) where the instance exists when the engine initializes. 
* engines like `tk-photoshop` which itself create their own `QApplication` instances
* `tk-shell` which creates the `QApplication` instance when a registered command is executed 

So we might be able to hit all these cases in the engine base class (which is why it's called 3 times) but I felt like there was enough variation in how engines worked that maybe it should be available for subclasses to call.

I was actually pondering making it public so that apps could call it just to ensure that the SG-approved font was available.

Anyway, that's my thought process there. Let me know what you think.